### PR TITLE
Fix compilation under -Ydebug

### PR DESCRIPTION
**Pull request by retronym**
Review by @xeno-by

Before this patch:

```
% scalac -Ydebug test/files/pos/t3999b.scala
..
warning: !!! ... hit cycle trying to complete class FT
test/files/pos/t3999b.scala:5: error: illegal cyclic reference involving class FT
  case class FT(f : Float) extends Score
             ^
locals (2 total):
  constructor FT
  value f
locals (6 total):
  class FT
  module FT
  value <local FT>
  value f
  constructor FT
  value f
```

### CODE REVIEW

**Comment by retronym** [comment]
PLS REBUILD ALL

```
http://private-repo.typesafe.com/typesafe/scala-pr-validation-snapshots/
[artifact:dependencies] Transferring 3K from extra-repo
[artifact:dependencies] Downloading: org/scala-lang/scala-library/2.11.0-c258ccc-SNAPSHOT/scala-library-2.11.0-c258ccc-20140117.175159-1.jar from repository extra-repo at http://private-repo.typesafe.com/typesafe/scala-pr-validation-snapshots/
[artifact:dependencies] Transferring 5472K from extra-repo
Build timed out (after 240 minutes). Marking the build as aborted.
Build was aborted
Notifying upstream projects of job completion
Finished: ABORTED
```

File: src/compiler/scala/tools/nsc/ast/Trees.scala [hunk 1]
```diff
@@ -194,7 +194,9 @@ trait Trees extends scala.reflect.internal.Trees { self: Global =>
    *  (bq:) This transformer has mutable state and should be discarded after use
    */
   private class ResetAttrs(localOnly: Boolean, leaveAlone: Tree => Boolean = null, keepLabels: Boolean = false) {
-    val debug = settings.debug.value
+    // this used to be based on -Ydebug, but the need for logging in this code is so situational
+    // that I've reverted to a hard-coded constant here.
+    val debug = false
```
Comment (xeno-by) [inline src/compiler/scala/tools/nsc/ast/Trees.scala:199]: Thanks

File: src/reflect/scala/reflect/internal/Symbols.scala [hunk 2]
```diff
@@ -2419,7 +2419,7 @@ trait Symbols extends api.Symbols { self: SymbolTable =>
         else if (isTrait) ("trait", "trait", "TRT")
         else if (isClass) ("class", "class", "CLS")
         else if (isType) ("type", "type", "TPE")
-        else if (isClassConstructor && isPrimaryConstructor) ("primary constructor", "constructor", "PCTOR")
+        else if (isClassConstructor && (owner.hasCompleteInfo && isPrimaryConstructor)) ("primary constructor", "constructor", "PCTOR")
```
Comment (xeno-by) [inline src/reflect/scala/reflect/internal/Symbols.scala:2422]: Nice catch!

**Comment by xeno-by** [comment]
LGTM

**Pull request merged**